File: App.client/projects/proctor-app/scripts/models/complaint-evidence.ts
import { User } from './user';

export class ComplaintEvidence {
  id: number;
  complaintId: number;
  filePath: string;
  uploadedBy: number;
  uploadedAt: string;
  description?: string | null;
  createdOn: string;
  modifiedOn: string;
  uploader?: User | null;

  constructor() {
    this.id = 0;
    this.complaintId = 0;
    this.filePath = '';
    this.uploadedBy = 0;
    this.uploadedAt = new Date().toISOString();
    this.description = null;
    this.createdOn = this.modifiedOn = '';
    this.uploader = null;
  }
}

File: App.Models/ComplaintEvidence.cs
using App.Core.Models;

namespace App.Models
{
    public class ComplaintEvidence : IAuditableEntity
    {
        public int Id { get; set; }
        public int ComplaintId { get; set; }
        public string FilePath { get; set; } = string.Empty;
        public int UploadedBy { get; set; }
        public DateTime UploadedAt { get; set; }
        public string? Description { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }

        // Navigation properties
        public virtual Complaint Complaint { get; set; } = null!;
        public virtual User Uploader { get; set; } = null!;
    }
}


